code/modules/spells/spell.dm

```diff
@@ -373,10 +373,18 @@
 			to_chat(owner, span_warning("You can't position your hands correctly to invoke [src]!"))
 		return FALSE
 
-	if((invocation_type == INVOCATION_WHISPER || invocation_type == INVOCATION_SHOUT) && !living_owner.can_speak())
-		if(feedback)
-			to_chat(owner, span_warning("You can't get the words out to invoke [src]!"))
-		return FALSE
+	if((invocation_type == INVOCATION_WHISPER || invocation_type == INVOCATION_SHOUT))
+		if(!HAS_TRAIT(living_owner,TRAIT_SIGN_LANG) && !living_owner.can_speak())
+			if(feedback)
+				to_chat(owner, span_warning("You can't get the words out to invoke [src]!"))
+			return FALSE
+		// Handle sign language
+		else if(HAS_TRAIT(living_owner, TRAIT_SIGN_LANG))
+			var/sigreturn = SEND_SIGNAL(living_owner, COMSIG_CARBON_TRY_SIGN_SPELL)
```

Comment:
in the signal documentation, you list the first argument as `mob/user`, which is incorrect, as you never pass that in. If you are talking about the `datum/source` that is passed in with every signal, that shouldn't be included in the argument documentation in `signals_spell.dm`, as it is implied

Reply:
Honestly speaking, I still don't know how to document signals properly, I was going off of what I was told would work for that. Would I just say it's going to datum/components/sign_language?